Deployment/ASP.Net_Core_to_EC2/Readme.md

```diff
@@ -3,215 +3,436 @@
-    our application.
+    ![View_Instance](./media/ec210.png)
 
-### Module 5: Integrate VSTS and AWS CodeDeploy
+### Module 7: Set up a Service role for AWS CodeDeploy
 
 #### Overview
 
-In this step we will create a VSTS build job that will deploy our
+In this step we will create a Service Role which will grant AWS CodeDeploy access to the target EC2 instance.
+
+#### Implementation Instructions
+
+1.  From the AWS Console search for IAM.
+1.  Select Roles > Create Role
+
+    ![New_Role](./media/new_role.png)
+
+1.  Select AWS service under Select type of trusted entity.
+1.  Select CodeDeploy under Choose a use case
+
+    ![New_Role](./media/service_role.png)
+
+1.  Select Next: Permissions
+
+    ![New_Role](./media/role.png)
+
+1.  A policy has already been attached. Select Next:Tags.
+
+    ![New_Role](./media/create_role.png)
+
+1.  Select Next: Review 
+
+    ![New_Role](./media/role_tag.png)
+
+1.  Enter Role name and select Create role
+
+    ![New_Role](./media/role_final.png)
+
+
+### Module 8: Setup AWS CodeDeploy
+
+#### Overview
+
+In this step we will create an application and a
+deployment group in CodeDeploy.
+
+#### Implementation Instructions
+
+1.  From the AWS Console, search for CodeDeploy.
+
+    ![codedeploy_search](./media/codedeploy.png)
+
+1.  Go to Applications, click Create application
+
+    ![get_started_now](./media/cd_app.png)
+
+1.  Enter the Application name
+1.  Select EC2/On-premises under Compute platform and select Create application
+
+    ![sample_deployment](./media/cd_create_app.png)
+
+1.  Go to the application and select Create deployment group
+
+    ![inplace_deployment](./media/cd_create_grp.png)
+
+1.  Enter the Deployment group name
+1.  Select the Service Role created in Module 7
+
+    ![windows_type](./media/cd_grp1.png)
+
+1.  Choose In-place deployment.
+1.  Select Amazon EC2 instances under Environment configuration
+1.  Add a tag with Key - Name, Value - (Name of the EC2 instance created in Module 6) This will identify the EC2 instance created in Module 6
+
+    ![select_windows_sample](./media/cd_grp2.png)
+
+1.  Select Never under Agent configuration with AWS Systems Manager as we have already configured the EC2 instance with the CodeDeploy Agent.
+1.  Select CodeDeployDefault.AllAtOnce under Deployment Settings.
+1.  Deselect Enable load balancing.
+1.  Select Create deployment group
+
+    ![service_role](./media/cdgrp3.png)
+
+1.  CodeDeploy is ready to receive our application.
+
+### Module 9: Create an Amazon S3 bucket
+
+#### Overview
+
+In this module, we will create an Amazon S3 bucket to which the revision bundle will be uploaded. A revision in AWS CodeDeploy is a version of an application you want to deploy. Our sample application revisions are stored in Amazon S3
+
+#### Implementation
+
+1. From the AWS Console search for S3.
+1. Select Create bucket.
+
+    ![s3_bucket](./media/s31.png)
+
+1. Enter the Bucket name and select the AWS Region
```

Comment:
I believe the bucket has to be in the same region as the CodeDeploy resources, worth checking and noting if so.